src/chatlist/chatList.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { withStyles } from "@material-ui/core/styles";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import ListItemAvatar from "@material-ui/core/ListItemAvatar";
+import Avatar from "@material-ui/core/Avatar";
+import Typography from "@material-ui/core/Typography";
+import styles from "./styles";
+import Divider from "@material-ui/core/Divider";
+import Button from "@material-ui/core/Button";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import NotificationImportant from "@material-ui/icons/NotificationImportant";
+
+class ChatListComponent extends React.Component {
```

Comment:
Just curious - I think I remember you mentioning that you preferred **functional** React components, over **class** based ones. Am I remembering correctly? If so, this would be a good opportunity to try converting this class component into a functional one.

Reply:
Yes, you're remembering it correctly. I would prefer to use functional components, but here I am using `componentDidMount` and `componentDidUpdate`. That's why I thought I should keep class-based ones.

Reply:
In that case, have you used `useEffect` yet? It's one of the new React hooks which let you do similar things to cDM and cDU but with functional components. For example;

```
useEffect(() => {
// the code in here will run once on mount only.
// notice the empty array on the line below. the next example you'll see what this is for
}, [])
```

And if you want it to run on every update;

```
useEffect(() => {
// the code in here will run once on mount.
// and it will run again whenever variableName changes. 
// you can put multiple things in the dependency array below, it doesnt just have to be one variable of course.
// basically any time something inside the dependency array changes, it will re-run this effect
}, [variableName])
```